src/errors.h

```diff
@@ -37,9 +37,19 @@ GIT_INLINE(int) git_error_set_after_callback_function(
 }
 
 #ifdef GIT_WIN32
+#if defined(_MSC_VER) && _MSC_VER >= 1300
 #define git_error_set_after_callback(code) \
 	git_error_set_after_callback_function((code), __FUNCTION__)
 #else
+/* VC++6 doesn't support __FUNCTION__, fallback to __FILE__:__LINE__, with
+ * macro to handle the fact _LINE__ isn't a string
+ */
+#define __AS_STRING__(x) #x
+#define __LINESTRING__ __AS_STRING__(__LINE__)
+#define git_error_set_after_callback(code) \
+	git_error_set_after_callback_function((code), __FILE__ ":" __LINESTRING__)
+#endif
+#else
```

Comment:
Since \_\_func\_\_ doesn't exist in C89 I'd personally prefer to have this one used for more cases and only use \_\_func\_\_ if defined(\_\_STDC_VERSION\_\_) && (\_\_STDC_VERSION\_\_ >= 199901L)

Considering that the tests uses \_\_func\_\_ a lot it could also be worth considering making a `#define __func__ __FUNCTION__` in a include file somewhere for the windows case, and maybe even have your fallback as an alternative for C89 compliance.